### PR TITLE
BDP-3878: Magnet on flow ramping regardless failure

### PR DESCRIPTION
**Pull request by kxu1026**

### CODE REVIEW

**Comment by kxu1026** [comment]
The requirement is ' If the workflow fails during the ramp, DONNOT stop ramping it and do not logging this failed flow into the exclude-ramp list.' I am leveraging the maxFailureToPause and maxFailureToRampDown configure to make it happen. When both maxFailureToPause and maxFailureToRampDown are set to negative value, the failed workflow will not be logged into the exclude-ramp list.

File: azkaban-common/src/main/java/azkaban/executor/ExecutableRamp.java [hunk 1]
```diff
@@ -376,6 +376,13 @@ public boolean isNotTestable() {
     return (!this.state.isActive || this.state.isPaused || (this.state.rampStage <= 0));
   }
 
+  /**
+   * Return true iif this ramp will not be required to put those failed workflow into the exclude-ramp list
+   */
+  public boolean isMagneticTest() {
```
Comment (zhouyejoe) [inline azkaban-common/src/main/java/azkaban/executor/ExecutableRamp.java:382]: Function name should be generic, maybe "ignoreFlowFailure".

File: azkaban-common/src/main/java/azkaban/executor/ExecutableRamp.java [hunk 2]
```diff
@@ -376,6 +376,13 @@ public boolean isNotTestable() {
     return (!this.state.isActive || this.state.isPaused || (this.state.rampStage <= 0));
   }
 
+  /**
+   * Return true iif this ramp will not be required to put those failed workflow into the exclude-ramp list
```
Comment (jakhani) [inline azkaban-common/src/main/java/azkaban/executor/ExecutableRamp.java:380]: nit: if

File: azkaban-common/src/main/java/azkaban/executor/ExecutableRamp.java (hunk 1 above)
Comment (jakhani) [inline azkaban-common/src/main/java/azkaban/executor/ExecutableRamp.java:382]: Shall this method name be changed? Magnetic is very specific to Spark right?

File: azkaban-common/src/main/java/azkaban/executor/ExecutableRamp.java (hunk 2 above)
Comment (kxu1026) [reply azkaban-common/src/main/java/azkaban/executor/ExecutableRamp.java:380]: iif = if and only if. right?

File: azkaban-common/src/main/java/azkaban/executor/ExecutableRamp.java (hunk 1 above)
Comment (kxu1026) [reply azkaban-common/src/main/java/azkaban/executor/ExecutableRamp.java:382]: what is your suggestion?

Comment (kxu1026) [reply azkaban-common/src/main/java/azkaban/executor/ExecutableRamp.java]: changed to ignoreTestFailure

File: azkaban-common/src/main/java/azkaban/executor/ExecutableRamp.java (hunk 2 above)
Comment (kxu1026) [reply azkaban-common/src/main/java/azkaban/executor/ExecutableRamp.java:380]: changed to if and only if

File: azkaban-common/src/main/java/azkaban/executor/ExecutableRamp.java [hunk 3]
```diff
@@ -376,6 +376,13 @@ public boolean isNotTestable() {
...
+   */
+  public boolean ignoreTestFailure() {
+    return (this.metadata.maxFailureToPause < 0) && (this.metadata.maxFailureToRampDown < 0);
```
Comment (jakhani) [inline azkaban-common/src/main/java/azkaban/executor/ExecutableRamp.java:383]: <= ?

**Review by jakhani — DISMISSED** [review]

**Review by djaiswal83 — CHANGES_REQUESTED** [review]
A minor typo. Rest looks good.

**Review by djaiswal83 — APPROVED** [review]
Thanks for addressing the comments.

**Pull request merged**